scripts/config_primary.sh

```diff
@@ -0,0 +1,165 @@
+install_mongo3
+
+
+#start mongod
+mongod --dbpath /var/lib/mongo/ --bind_ip 0.0.0.0 --logpath /var/log/mongodb/mongod.log --fork
+
+sleep 30
+n=`ps -ef |grep "mongod --dbpath /var/lib/mongo/"|grep -v grep | wc -l`
+if [[ $n -eq 1 ]];then
+    echo "mongod started successfully"
+else
+    echo "mongod started failed!"
+fi
+
+#create users
+mongo <<EOF
+use admin
+db.createUser({user:"$mongoAdminUser",pwd:"$mongoAdminPasswd",roles:[{role: "userAdminAnyDatabase", db: "admin" },{role: "readWriteAnyDatabase", db: "admin" },{role: "root", db: "admin" }]})
+exit
+EOF
+if [[ $? -eq 0 ]];then
+    echo "mongo user added succeefully."
+else
+    echo "mongo user added failed!"
+fi
+
+#stop mongod
+sleep 15
+MongoPid=`ps -ef |grep "mongod --dbpath /var/lib/mongo/"|grep -v grep |awk '{print $2}'`
+kill -2 $MongoPid
+
+
+
+#set keyfile
+echo "vfr4CDE1" > /etc/mongokeyfile
+chown mongod:mongod /etc/mongokeyfile
+chmod 600 /etc/mongokeyfile
+sed -i 's/^#security/security/' /etc/mongod.conf
+sed -i '/^security/akeyFile: /etc/mongokeyfile' /etc/mongod.conf
+sed -i 's/^keyFile/  keyFile/' /etc/mongod.conf
+
+sleep 15
+MongoPid1=`ps -ef |grep "mongod --dbpath /var/lib/mongo/"|grep -v grep |awk '{print $2}'`
+if [[ -z $MongoPid1 ]];then
+    echo "shutdown mongod successfully"
+else
+    echo "shutdown mongod failed!"
+    kill $MongoPid1
+    sleep 15
+fi
+
+#restart mongod with auth and config replica set
+mongod --configsvr --replSet crepset --bind_ip 0.0.0.0 --port 27019 --dbpath /var/lib/mongo/ --logpath /var/log/mongodb/config.log --fork --config /etc/mongod.conf --keyFile /etc/mongokeyfile --sslMode requireSSL --sslPEMKeyFile /etc/MongoAuthCert.pem --sslPEMKeyPassword Mongo123
+
+
+#initiate config replica set
+
+for((i=1;i<=3;i++))
+do
+    sleep 15
+    n=`ps -ef |grep "mongod --configsvr"|grep -v grep |wc -l`
+    if [[ $n -eq 1 ]];then
+        echo "mongo config replica set started successfully"
+        break
+    else
+        mongod --configsvr --replSet crepset --bind_ip 0.0.0.0 --port 27019 --dbpath /var/lib/mongo/ --logpath /var/log/mongodb/config.log --fork --config /etc/mongod.conf --keyFile /etc/mongokeyfile --sslMode requireSSL --sslPEMKeyFile /etc/MongoAuthCert.pem --sslPEMKeyPassword Mongo123
+        continue
+    fi
+done
+
+n=`ps -ef |grep "mongod --configsvr"|grep -v grep |wc -l`
+if [[ $n -ne 1 ]];then
+    echo "mongo config replica set tried to start 3 times but failed!"
+fi
+
+
+mongo --port 27019 --ssl --sslCAFile /etc/MongoAuthCert.pem --sslAllowInvalidHostnames<<EOF
+use admin
+db.auth("$mongoAdminUser","$mongoAdminPasswd")
+config={_id: "crepset", configsvr: true, members: [{ _id: 0, host: "10.0.0.240:27019" },{ _id: 1, host: "10.0.0.241:27019" },{ _id: 2, host: "10.0.0.242:27019" }]}
+rs.initiate(config)
+exit
+EOF
+if [[ $? -eq 0 ]];then
+    echo "mongod config replica set initiated succeefully."
+else
+    echo "mongod config replica set initiated failed!"
+fi
+
+
+
+#set mongod auto start
+cat > /etc/init.d/mongod1 <<EOF
```

Comment:
We should move the start/stop/restart logic into a new script as it is duplicated in all scripts